File: solutions/java/bob/1/src/main/java/Bob.java
public class Bob {
    String hey(String phrase) {
        String trimmedPhrase = phrase.trim();
        if (trimmedPhrase.isEmpty()) return "Fine. Be that way!";
        if (isUpper(trimmedPhrase) && trimmedPhrase.endsWith("?")) return "Calm down, I know what I'm doing!";
        if (isUpper(trimmedPhrase)) return "Whoa, chill out!";
        if (trimmedPhrase.endsWith("?")) return "Sure.";
        return "Whatever.";
    }

    boolean isUpper(String s) {
        return s.chars().anyMatch(Character::isLetter) && s.equals(s.toUpperCase());
    }
}

File: solutions/csharp/bob/1/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        if (statement.TrimEnd().EndsWith("?"))
        {
            if (statement.ToUpper().Equals(statement) && statement.Any(c => char.IsLetter(c)))
                { return "Calm down, I know what I'm doing!"; }

            return "Sure.";
        }
        else if (statement.ToUpper().Equals(statement) && statement.Length > 0 && statement.Any(c => char.IsLetter(c))) 
                { return "Whoa, chill out!"; }
        else if (statement.Trim().Length == 0) { return "Fine. Be that way!"; }
        return "Whatever.";
    }
}